### PR TITLE
Preserve spacing on trivia for invert if

### LINKED ISSUE #42715: "Invert if" removes newlines
**Steps to Reproduce**:

``` C#
DateTime lastFileWriteTime = File.GetLastWriteTimeUtc(path);

if (lastFileWriteTime == s_minFileTime) // <-- Invert if
{
    return null;
}

return lastFileWriteTime;
```

**Expected Behavior**:
``` C#
DateTime lastFileWriteTime = File.GetLastWriteTimeUtc(path);

if (lastFileWriteTime != s_minFileTime)
{
   return lastFileWriteTime;
}

return null;
```

**Actual Behavior**:

``` C#
DateTime lastFileWriteTime = File.GetLastWriteTimeUtc(path);

if (lastFileWriteTime != s_minFileTime)
{
    return lastFileWriteTime;
}
return null;
```

### PR DESCRIPTION
**Pull request by ryzngard**
Fixes #42715

### CODE REVIEW

File: src/Features/Core/Portable/InvertIf/AbstractInvertIfCodeRefactoringProvider.cs [hunk 1]
```diff
@@ -472,10 +475,26 @@ private ImmutableArray<StatementRange> GetSubsequentStatementRanges(TIfStatement
...
+                        // {
+                        //    return true;
+                        // }
+                        //              // <<< preserve this line
+                        // // preserve this comment
+                        // return false;
+                        var leadingTrivia = GetLeadingSpace(statementsAfterIf[0].GetLeadingTrivia()).Concat(GetTriviaAfterSpace(newTrailing[0].GetLeadingTrivia()));
+                        var trailingTrivia = GetTriviaUntilSpace(newTrailing[^1].GetTrailingTrivia()).Concat(GetTrailingSpace(statementsAfterIf[^1].GetTrailingTrivia()));
+                        newTrailing[0] = newTrailing[0].WithLeadingTrivia(leadingTrivia);
+                        newTrailing[^1] = newTrailing[^1].WithTrailingTrivia(trailingTrivia);
```
Comment (CyrusNajmabadi) [inline src/Features/Core/Portable/InvertIf/AbstractInvertIfCodeRefactoringProvider.cs:496]: unclear how you know that newTrailing has at least oen element in it.

**Review by CyrusNajmabadi — APPROVED** [review]

File: src/Features/Core/Portable/InvertIf/AbstractInvertIfCodeRefactoringProvider.cs (hunk 1 above)
Comment (ryzngard) [reply src/Features/Core/Portable/InvertIf/AbstractInvertIfCodeRefactoringProvider.cs:496]: It was previously the body of the if statement. Let me see if there are tests to verify my assumption though. I don't think we'd make it here if it was empty

Comment (CyrusNajmabadi) [reply src/Features/Core/Portable/InvertIf/AbstractInvertIfCodeRefactoringProvider.cs:496]: Can you add a test where there is no trivia anywhere in the if and the statements?

Comment (CyrusNajmabadi) [reply src/Features/Core/Portable/InvertIf/AbstractInvertIfCodeRefactoringProvider.cs]: (like no spaces/newlines *at all*).

Comment (ryzngard) [reply src/Features/Core/Portable/InvertIf/AbstractInvertIfCodeRefactoringProvider.cs]: oh you mean like...

```csharp
if(true){return true;}return false;
```

Comment (CyrusNajmabadi) [reply src/Features/Core/Portable/InvertIf/AbstractInvertIfCodeRefactoringProvider.cs]: Yup, exactly.  :)

Comment (CyrusNajmabadi) [reply src/Features/Core/Portable/InvertIf/AbstractInvertIfCodeRefactoringProvider.cs]: or even better:

```
{if(true){return(true);}return(false);}
```

No spaces/newlines anywhere.

Comment (ryzngard) [reply src/Features/Core/Portable/InvertIf/AbstractInvertIfCodeRefactoringProvider.cs]:
```csharp
return(true);
```

oh no...

**Pull request merged**